src/consensus/pbft/libbyz/pre_prepare.cpp

```diff
@@ -474,4 +474,23 @@ const std::array<uint8_t, MERKLE_ROOT_SIZE>& Pre_prepare::
 int64_t Pre_prepare::get_ctx() const
 {
   return rep().ctx;
+}
+
+std::vector<int> Pre_prepare::get_prev_pp_valid_principals()
+{
+  if (rep().num_prev_pp_sig == 0)
+  {
+    return {};
+  }
+  std::vector<int> principals_with_proof;
```

Comment:
you can pre-allocate this vector

Reply:
I wouldn't worry about this on this PR as it will change on the next one where we will also be verifying the proofs

Reply:
This function is used in one place and can just be combined with said function